runtime/recipe/view-connection.js

```diff
@@ -58,14 +60,29 @@ class ViewConnection {
   get recipe() { return this._recipe; }
   get name() { return this._name; } // Parameter name?
   get tags() { return this._tags; }
-  get type() { return this._type; }
+  get type() {
+    if (this._type)
+      return this._type;
+    return this._rawType;
+  }
+  get rawType() {
```

Comment:
`raw` is a bit terse, but maybe I can get over that... unresolved/original/underlying